### PR TITLE
Feature/lockdown

### PR DESCRIPTION
**Pull request by chaosfreak93**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -77,6 +79,33 @@ client.on('clickMenu', async (menu) => {
...
+    }, 1000);
+}
+
+async function checkLockdownChannel() {
+    let lockdownChannel: ILockdown[] = (await DatabaseService.getData()).lockdownChannel;
+
+    await lockdownChannel.forEach(async (lockDownData) => {
+        if (Date.now() > lockDownData.until) {
+            const channel = client.channels.cache.get(lockDownData.channelId) as Discord.TextChannel;
+            const role = client.guilds.cache.get("872092708527833139").roles.everyone;
```
Comment (Stuyk) [inline src/index.ts:94]: Use `getGuild` from the index here.

File: src/index.ts [hunk 2]
```diff
@@ -77,6 +79,33 @@ client.on('clickMenu', async (menu) => {
     await menu.reply.edit("Roles were updated successfully.");
 });
 
+function handleTick() {
+    setInterval(async () => {
+        await checkLockdownChannel();
+    }, 1000);
+}
+
+async function checkLockdownChannel() {
```
Comment (Stuyk) [inline src/index.ts:88]: This should be moved into the `lockdown` command itself and add an export to target it.

File: src/commands/lockdown.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { ILockdown } from '../interfaces/ILockdown';
+import { DatabaseService } from '../service/database';
+import RegexUtility from '../utility/regex';
+
+
+const command: ICommand = {
+    command: 'lockdown',
+    description: '<channel> <minutes> - Locks a Channel.',
+    execute: async (msg: Discord.Message, id: string, minutes: number) => {
+        const channel = msg.guild.channels.cache.find(ch => ch.id == id || ch.id == RegexUtility.parseChannelID(id)) as Discord.TextChannel;
```
Comment (Stuyk) [inline src/commands/lockdown.ts:13]: Parse the channel id first with regex; and then perform a `cache.get`. No need for find here.

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: src/commands/lockdown.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+export async function checkLockdownChannel() {
+    let lockdownChannel: ILockdown[] = (await DatabaseService.getData()).lockdownChannel;
+
+    lockdownChannel.forEach(async (lockDownData) => {
+        if (Date.now() > lockDownData.until) {
+            const guild = getGuild();
+            const role = guild.roles.everyone;
+            const channel = guild.channels.cache.get(lockDownData.channelId) as Discord.TextChannel;
+            if (!channel.permissionsFor(role).has('SEND_MESSAGES')) {
+                await channel.updateOverwrite(role,{'SEND_MESSAGES': true});
```
Comment (Stuyk) [inline src/commands/lockdown.ts:67]: Is there a way to set this to neutral instead of `true`? It bugs permission(s) if it is not neutral.

**Review by Stuyk — CHANGES_REQUESTED** [review]

File: src/commands/lockdown.ts (hunk 4 above)
Comment (chaosfreak93) [reply src/commands/lockdown.ts:67]: At the moment, I don't know any way. I will look for a solution

**Review by Stuyk — APPROVED** [review]

**Pull request merged**